syft/tasks.go

```diff
@@ -0,0 +1,307 @@
+
+		// OS package declared catalogers
+		newTaskDescriptor(cfg, rpm.NewFileCataloger(), declaredTag, directoryTag, packageTag, osTag, "rpm", "redhat"),
+
+		// language-specific package installed catalogers
+		newTaskDescriptor(cfg, dotnet.NewDotnetDepsCataloger(), installedTag, imageTag, packageTag, languageTag, "dotnet", "c#"),
+		newTaskDescriptor(cfg, javascript.NewJavascriptPackageCataloger(), installedTag, imageTag, packageTag, languageTag, "javascript", "node"),
+		newTaskDescriptor(cfg, php.NewPHPComposerInstalledCataloger(), installedTag, imageTag, packageTag, languageTag, "php", "composer"),
+		newTaskDescriptor(cfg, ruby.NewGemSpecCataloger(), installedTag, imageTag, packageTag, languageTag, "ruby", "gem"),
+		newTaskDescriptor(cfg, rust.NewCargoLockCataloger(), installedTag, imageTag, packageTag, languageTag, "rust", "binary"),
+
+		// language-specific package declared catalogers
+		newTaskDescriptor(cfg, cpp.NewConanCataloger(), declaredTag, directoryTag, packageTag, languageTag, "cpp", "conan"),
+		newTaskDescriptor(cfg, dart.NewPubspecLockCataloger(), declaredTag, directoryTag, packageTag, languageTag, "dart"),
+		newTaskDescriptor(cfg, dotnet.NewDotnetDepsCataloger(), declaredTag, directoryTag, packageTag, languageTag, "dotnet", "c#"),
+		newTaskDescriptor(cfg, haskell.NewHackageCataloger(), declaredTag, directoryTag, packageTag, languageTag, "haskell", "cabal"),
+		newTaskDescriptor(cfg, golang.NewGoModFileCataloger(), declaredTag, directoryTag, packageTag, languageTag, "go", "golang", "gomod"),
+		newTaskDescriptor(cfg, java.NewJavaPomCataloger(), declaredTag, directoryTag, packageTag, languageTag, "java", "maven"),
+		newTaskDescriptor(cfg, javascript.NewJavascriptLockCataloger(), declaredTag, directoryTag, packageTag, languageTag, "javascript", "node"),
+		newTaskDescriptor(cfg, php.NewPHPComposerLockCataloger(), declaredTag, directoryTag, packageTag, languageTag, "php", "composer"),
+		newTaskDescriptor(cfg, python.NewPythonIndexCataloger(), declaredTag, directoryTag, packageTag, languageTag, "python"),
+		newTaskDescriptor(cfg, ruby.NewGemFileLockCataloger(), declaredTag, directoryTag, packageTag, languageTag, "ruby", "gem"),
+		newTaskDescriptor(cfg, rust.NewCargoLockCataloger(), declaredTag, directoryTag, packageTag, languageTag, "rust", "cargo"),
+		newTaskDescriptor(cfg, swift.NewCocoapodsCataloger(), declaredTag, directoryTag, packageTag, languageTag, "swift", "cocoapods"),
+
+		// language-specific package declared & installed catalogers
+		newTaskDescriptor(cfg, python.NewPythonPackageCataloger(), directoryTag, installedTag, imageTag, packageTag, languageTag, "python"),
+		newTaskDescriptor(cfg, golang.NewGoModuleBinaryCataloger(), directoryTag, installedTag, imageTag, packageTag, languageTag, "go", "golang", "gomod", "binary"),
+		newTaskDescriptor(cfg, java.NewJavaCataloger(cfg.Java()), directoryTag, installedTag, imageTag, packageTag, languageTag, "java", "maven"),
```

Comment:
note: we should consider renaming the `java-cataloger` to `java-archive-cataloger`